src/coreclr/debug/createdump/dumpwriter.cpp

```diff
@@ -203,23 +203,23 @@ DumpWriter::WriteDump()
         // Only write the regions that are backed by memory
         if (memoryRegion.IsBackedByMemory())
         {
-            uint32_t size = memoryRegion.Size();
+            uint64_t size = memoryRegion.Size();
             uint64_t address = memoryRegion.StartAddress();
             total += size;
 
             while (size > 0)
             {
-                uint32_t bytesToRead = std::min(size, (uint32_t)sizeof(m_tempBuffer));
+                size_t bytesToRead = std::min(size, sizeof(m_tempBuffer));
```

Comment:
size should probably be decleared above as `size_t` ... Otherwise
```suggestion
                size_t bytesToRead = std::min((size_t)size, sizeof(m_tempBuffer));
```
```console
2021-03-11T02:11:41.5645060Z /Users/runner/work/1/s/src/coreclr/debug/createdump/dumpwriter.cpp:212:38: error: no matching function for call to 'min'
2021-03-11T02:11:41.5744730Z                 size_t bytesToRead = std::min(size, sizeof(m_tempBuffer));
2021-03-11T02:11:41.5845290Z                                      ^~~~~~~~
2021-03-11T02:11:41.5947840Z /Applications/Xcode_12.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:2560:1: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('unsigned long long' vs. 'unsigned long')
2021-03-11T02:11:41.6050380Z min(const _Tp& __a, const _Tp& __b)
2021-03-11T02:11:41.6152150Z ^
2021-03-11T02:11:41.6255530Z /Applications/Xcode_12.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:2571:1: note: candidate template ignored: could not match 'initializer_list<type-parameter-0-0>' against 'unsigned long long'
2021-03-11T02:11:41.6357920Z min(initializer_list<_Tp> __t, _Compare __comp)
2021-03-11T02:11:41.6460080Z ^
2021-03-11T02:11:41.6561170Z /Applications/Xcode_12.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:2551:1: note: candidate function template not viable: requires 3 arguments, but 2 were provided
2021-03-11T02:11:41.6662240Z min(const _Tp& __a, const _Tp& __b, _Compare __comp)
2021-03-11T02:11:41.6761820Z ^
2021-03-11T02:11:41.6863770Z /Applications/Xcode_12.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:2580:1: note: candidate function template not viable: requires single argument '__t', but 2 arguments were provided
2021-03-11T02:11:41.6993010Z min(initializer_list<_Tp> __t)
2021-03-11T02:11:41.7195180Z ^
2021-03-11T02:11:42.0217020Z 1 error generated.
```